src/test/integration/users/User.t.sol

```diff
@@ -275,9 +297,35 @@ contract User is Logger, IDelegationManagerTypes, IAllocationManagerTypes {
         delegationManager.undelegate(address(staker));
         print.gasUsed();
 
+        // Operator is calling this function themselves
+        // handle the undelegating staker on this contract to manage the delegatedStakers enum set
+        handleUndelegatingStaker(address(staker));
+
         return expectedWithdrawals;
     }
 
+    /// @dev NOTE: do NOT call this function directly. This is simply implemented to manage the
+    /// delegatedStakers enum set. This function is only meant to be called on the Operator User contract.
+    function handleDelegatingStakers(address staker) public virtual {
+        // check that the current User is an operator
+        require(
+            delegationManager.isOperator(address(this)),
+            "User is not an operator"
+        );
```

Comment:
Let's leave the checks out of this. If you're failing these, you'll see failures elsewhere - so let's keep these methods as simple as possible.